### PR TITLE
IMPRO-2091: Remove mid-point option for coordinate expansion

### PR DESCRIPTION
**Pull request by bayliffe**
This change removes the use_midpoint option from the combine CLI, the cube combiner, and associated functionality in the expand_bounds function within cube_manipulation. Using this functionality with time coordinates (e.g. constructing accumulations) is not consistent with our metadata standards. No other use cases were found.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: improver/cli/combine.py [hunk 1]
```diff
@@ -86,8 +84,6 @@ def process(
         )
 
     else:
-        result = CubeCombiner(operation)(
-            CubeList(cubes), new_name, use_midpoint=use_midpoint,
-        )
+        result = CubeCombiner(operation)(CubeList(cubes), new_name)
```
Comment (gavinevans) [inline improver/cli/combine.py:87]: Sorry this isn't related to your changes but I notice that the `check_metadata` argument in this CLI is unused. Could you remove this argument and associated docstring?

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks for these clean-up, @bayliffe 👍 

I've just added one tangential comment.

File: improver/cli/combine.py (hunk 1 above)
Comment (bayliffe) [reply improver/cli/combine.py:87]: Done. I also checked no apps in the suite are providing this redundant option; they aren't.

**Review by gavinevans — DISMISSED** [review]

**Review by MoseleyS — DISMISSED** [review]
Looks fine to me.

**Comment by MoseleyS** [comment]
Waiting for #1448 to be merged first as minor conflicts are expected.

**Review by gavinevans — APPROVED** [review]
Thanks @bayliffe, this PR seems fine following the incorporation of the changes from #1448 👍

**Pull request merged**